.github/workflows/generate-analysis.yml

```
@@ -0,0 +1,224 @@
+name: Generate analysis files
+
+on:
+  workflow_call:
+    inputs:
+      from_version:
+        required: true
+        type: string
+      from_version_requirements:
+        default: 'gevent==22.10.2 greenlet==2.0.2'
+        type: string
+      from_version_requirements_sed:
+        default: '/^gevent\>/d; /^greenlet\>/d'
+        type: string
+      to_version:
+        required: true
+        type: string
+      to_version_requirements:
+        default: 'gevent==22.10.2 greenlet==2.0.2'
+        type: string
+      to_version_requirements_sed:
+        default: '/^gevent\>/d; /^greenlet\>/d'
+        type: string
+      python_version:
+        default: "3.10"
+        type: string
+      postgres_version:
+        default: "14"
+        type: string
+
+jobs:
+  generate-analysis:
+    name: Generate analysis ${{ inputs.to_version }}
+    runs-on: ubuntu-latest
+    env:
+      PGHOST: "localhost"
+      PGPASSWORD: "odoo"
+      PGUSER: "odoo"
+      FROM_VERSION: "${{ inputs.from_version }}"
+      TO_VERSION: "${{ inputs.to_version }}"
+    steps:
+      - name: Configure Postgres
+        uses: harmon758/postgresql-action@v1
```

```suggestion
        uses: harmon758/postgresql-action@0be19fa37850b22cb4c9bbf28a03abbf44abd863 #v1
```
To limit supply-chain attacks

https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

See also the recent "tj-actions" affair.

---

there is also `peter-evans/create-pull-request@v7` line 215

---

thanks, you might be interested in https://github.com/OCA/OpenUpgrade/pull/4901